src/App.vue

```diff
@@ -16,121 +17,165 @@
 </template>
 
 <script>
-import chroma from 'chroma-js';
-import Map from './Map';
+import chroma from "chroma-js";
+import Map from "./Map";
 import {
   getDynamicMapCss,
   getBaseCss,
-  getCombinedCssString,
-} from './dynamic-map-css';
+  getCombinedCssString
+} from "./dynamic-map-css";
 
 let legend = {
   data: null,
   code: null,
   name: null
-}
+};
 
 let position = {
   left: 0,
-  top: 0,
-}
+  top: 0
+};
 
 export default {
   name: "MapChart",
   components: { Map },
   watch: {
     countryData() {
       this.renderMapCSS();
-    },
+    }
   },
   props: {
     lowColor: {
       type: String,
-      default: '#fde2e2',
+      default: "#fde2e2"
     },
     highColor: {
       type: String,
-      default: '#d83737',
+      default: "#d83737"
+    },
+    chromaScaleOn: {
+      type: Boolean,
+      default: true
     },
     countryData: {
       type: Object,
-      required: true,
+      required: true
     },
     defaultCountryFillColor: {
       type: String,
-      default: '#dadada',
+      default: "#dadada"
     },
     countryStrokeColor: {
       type: String,
-      default: '#909090',
+      default: "#909090"
+    },
+    legendHeaderBackgroundColor: {
+      type: String,
+      default: "white"
     },
+
+    legendContentBackgroundColor: {
+      type: String,
+      default: "#dadbda8f"
+    },
+    legendFontColorHeader: {
+      type: String,
+      default: "black"
+    },
+    legendFontColorContent: {
+      type: String,
+      default: "black"
+    },
+    legendBorderColor: {
+      type: String,
+      default: "gray"
+    },
+    legendBorderRadius: {
+      type: Number,
+      default: 5
+    },
+    legendBoxShadow: {
+      type: Boolean,
+      default: false
+    }
   },
   data() {
     return {
       legend: legend,
       position: position,
-      node: document.createElement('style'),
-      chromaScale: chroma.scale(
-        [this.$props.lowColor, this.$props.highColor]
-      ),
+      node: document.createElement("style"),
+      chromaScale: chroma.scale([
+        this.$props.lowColor,
+        this.$props.highColor,
+        this.$props.defaultCountryFillColor,
+        this.$props.legendHeaderBackgroundColor,
+        this.$props.legendContentBackgroundColor,
+        this.$props.legendFontColorHeader,
+        this.$props.legendFontColorContent,
+        this.$props.legendBorderColor,
+        this.$props.legendBorderRadius,
+        this.$props.legendBoxShadow
```

Comment:
I think this is the problem. The **chroma.scale** has to contain two values, the lowColor and highColor (https://github.com/gka/chroma.js/).